PGO Training Fails with Error Binding to CoreCLR

PGO training on Windows fails with the following error:

```
Failed to load the dll from [D:\src\dotnet-optimization\output\CLRx64WINmas\PGO\cli\x64\shared\Microsoft.NETCore.App\3.0.0-preview6-27727-02\coreclr.dll], HRESULT: 0x8007007E
Failed to bind to CoreCLR at 'D:\src\dotnet-optimization\output\CLRx64WINmas\PGO\cli\x64\shared\Microsoft.NETCore.App\3.0.0-preview6-27727-02\'
Failed to create CoreCLR, HRESULT: 0x80008088
```

Git bisect points at #24619 as being the first PR containing the break.  It's not clear to me what in this change broke things, but in addition to the error above, there is also a restore error during CoreCLR layout generation that doesn't occur when things work.  The error comes from  ```build-test.cmd x64 release skipmanaged skipnative```:

```
D:\src\dotnet-optimization\output\CLRx64WINmas\PGO\coreclr-master\src\.nuget\optdata\optdata.csproj : error NU1101: Unable to find package optimization.PGO.CoreCLR. No packages exist with this id in source(s): https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json, https://dotnet.myget.org/F/sourcelink/api/v3/index.json, https://www.myget.org/F/xunit/api/v3/index.json, nuget.org [D:\src\dotnet-optimization\output\CLRx64WINmas\PGO\coreclr-master\src\build.proj]
D:\src\dotnet-optimization\output\CLRx64WINmas\PGO\coreclr-master\src\.nuget\optdata\optdata.csproj : error NU1101: Unable to find package optimization.IBC.CoreCLR. No packages exist with this id in source(s): https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json, https://dotnet.myget.org/F/sourcelink/api/v3/index.json, https://www.myget.org/F/xunit/api/v3/index.json, nuget.org [D:\src\dotnet-optimization\output\CLRx64WINmas\PGO\coreclr-master\src\build.proj]
    13 Warning(s)
    2 Error(s)
```

These packages are used for optimizing builds, not generating instrumented builds for producing training data, so I am suspicious that something associated with ```build.cmd pgoinstrument``` is broken.  This command should produce an instrumented build of CoreCLR that generates PGO data for each application run.

If you would like to repro this on your local machine, I can provide you with instructions offline.

--- a/src/.nuget/optdata/optdata.csproj
+++ b/src/.nuget/optdata/optdata.csproj
@@ -11,6 +11,13 @@
     <PackageReference Include="optimization.IBC.CoreCLR" Version="$(optimizationIBCCoreCLRVersion)" Condition="'$(optimizationIBCCoreCLRVersion)'!=''" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <RestoreSources>
+      https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json;
+      $(RestoreSources);
+    </RestoreSources>
+  </PropertyGroup>
+
   <Target Name="DumpPgoDataPackageVersion">
     <Message Importance="high" Text="$(optimizationPGOCoreCLRVersion)" />
   </Target>

